connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -395,6 +396,18 @@ message CachedLocalRelation {
   string hash = 3;
 }
 
+// Represents a remote relation that has been cached on server.
+message CachedRemoteRelation {
+  // (Required) An identifier of the user which cached the relation
+  string userId = 1;
+
+  // (Required) An identifier of the Spark session in which the relation is cached
+  string sessionId = 2;
```

Comment:
The user, session ID can't be trusted coming from the proto. THe cached relation must only have the actual unique ID of the relation ID and the rest is resolved from the context of the query.